pkg/authorization/local_authorizer.go

```diff
@@ -67,12 +68,15 @@ func (a *LocalAuthorizer) Authorize(ctx context.Context, attr authorizer.Attribu
 
 	reqScope := cluster.Name
 	filteredInformer := rbacwrapper.FilterInformers(reqScope, a.versionedInformers.Rbac().V1())
+	bootstrapInformer := rbacwrapper.FilterInformers(genericcontrolplane.LocalAdminCluster, a.versionedInformers.Rbac().V1())
+
+	mergedInformer := rbacwrapper.MergedInformers(filteredInformer, bootstrapInformer)
 
 	scopedAuth := rbac.New(
-		&rbac.RoleGetter{Lister: filteredInformer.Roles().Lister()},
-		&rbac.RoleBindingLister{Lister: filteredInformer.RoleBindings().Lister()},
-		&rbac.ClusterRoleGetter{Lister: filteredInformer.ClusterRoles().Lister()},
-		&rbac.ClusterRoleBindingLister{Lister: filteredInformer.ClusterRoleBindings().Lister()},
+		&rbac.RoleGetter{Lister: mergedInformer.Roles().Lister()},
+		&rbac.RoleBindingLister{Lister: mergedInformer.RoleBindings().Lister()},
```

Comment:
I've been thinking about this over the weekend and I'm not sure concatenating rules makes sense (at least I can't envision a problem here). 

The main risk in this approach is a name collision. Someone might accidentally use the same Role (or Role Binding) name in both, the local and the bootstrap authorizers. But we should be on the safe side here as we've the MergedInformers respect the order of the informers passed into them. In other words, you're always resolve things in the same order.

Reply:
IMO it is strange behaviour if we don't merge the rules, especially because it is non-continuous (in the mathematical sense). Imagine local bootstrap policy has Role X. Then X appears in a nother policy. Suddenly a user loses permissions. That's bad behaviour for an add operation.

It also plays badly with eventual consistency if you depend on that behaviour that permissions are removed.

Reply:
Yes, I see it now. Merging (Cluster) Rules is the key here.

Reply:
can we remove the merged binding lister?

Reply:
Done